src/common/selectors.js

```diff
@@ -448,6 +448,16 @@ export const createGetTags = collectionSelectors => {
   return _extendCollectionSelector(getTags, 'tag')
 }
 
+export const getLastShutdownTime = create(
+  (_, vm) => vm.id,
+  state => state.objects.byType.message,
+  (vmID, messages) => {
+    const vmLogsSD = filter(messages, message => message.$object === vmID && message.name === 'VM_SHUTDOWN')
+    .sort((message1, message2) => message1.time < message2.time)
+    return vmLogsSD.length !== 0 ? vmLogsSD[0].time : null
```

Comment:
It would be much efficient to iterate over the logs looking for the max time of `VM_SHUTDOWN` event.